src/libraries/System.Linq/tests/Shuffler.cs

```diff
@@ -11,13 +11,9 @@ public static class Shuffler
         public static T[] Shuffle<T>(T[] array)
         {
             var r = new Random(42);
-            int i = array.Length;
-            while (i > 1)
-            {
-                int j = r.Next(i--);
-                (array[i], array[j]) = (array[j], array[i]);
-            }
-            return array;
+            T[] copy = array.AsSpan().ToArray();
+            r.Shuffle(copy);
+            return copy;
```

Comment:
This had been mutating the `array` that was passed in and now it's making a copy.  How come?

Reply:
Hm, I think it returning the array threw me since I would have expected it to return `void`.  I'll undo that.